sw/device/silicon_creator/lib/drivers/rstmgr.c

```diff
@@ -0,0 +1,38 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "sw/device/silicon_creator/lib/drivers/rstmgr.h"
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "rstmgr_regs.h"
+
+enum {
+  kBase = TOP_EARLGREY_RSTMGR_AON_BASE_ADDR,
+};
+
+rstmgr_alert_info_t rstmgr_alert_info;
+
+static void rstmgr_alert_info_collect(void) {
+  rstmgr_alert_info.length =
+      abs_mmio_read32(kBase + RSTMGR_ALERT_INFO_ATTR_REG_OFFSET);
```

Comment:
nit: `& 0xF` or check if there is a static assert that could be used against the register field mask and the info array size.

Reply:
Fixed.